pilot/pkg/kube/inject/inject.go

```diff
@@ -187,15 +187,27 @@ func validateCIDRList(cidrs string) error {
 	return nil
 }
 
-func validatePortList(ports string) error {
-	if len(ports) > 0 {
-		for _, port := range strings.Split(ports, ",") {
-			if _, err := strconv.ParseInt(port, 10, 16); err != nil {
-				return fmt.Errorf("failed parsing port '%s': %v", port, err)
+func splitPorts(portsString string) []string {
+	return strings.Split(portsString, ",")
+}
+
+func parsePorts(portsString string) ([]int, error) {
+	if len(portsString) > 0 {
+		ports := make([]int, len(portsString))
```

Comment:
is this right? seems like the string has form 1000,2000 etc.
in general i'd advise not to bother with pre-sizing slices for small stuff like this, just use append to avoid any indexing problems.

Reply:
nope, it's not right ... fixed.